app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -9,16 +9,15 @@ def google_oauth2
 
     raw_info = request.env["omniauth.auth"]["extra"]["raw_info"]
     email = raw_info["email"]
-    @user = User.find_by_email(email)
-    unless @user
-      @user = User.create(email: email,
-           password: Devise.friendly_token[0,20])
+    name = request.env["omniauth.auth"]["info"]["name"]
+    Player.find_or_create_by(name: name, email: email)
```

Comment:
`find_or_create_by` is a neat helper provided by `ActiveRecord::Base`, and that ensures the existence of a `Player` for the logged in user with one line.